lib/aws/active_job/sqs/configuration.rb

```diff
@@ -17,20 +19,46 @@ class Configuration
           excluded_deduplication_keys: ['job_id']
         }.freeze
 
-        # @api private
-        attr_accessor :queues, :max_messages, :visibility_timeout,
-                      :shutdown_timeout, :client, :logger,
-                      :async_queue_error_handler, :message_group_id
+        GLOBAL_ENV_CONFIGS = %i[
+          config_file threads backpressure
+          max_messages shutdown_timeout
+          visibility_timeout message_group_id
+        ].freeze
 
-        attr_reader :excluded_deduplication_keys
+        QUEUE_ENV_CONFIGS = %i[
+          url max_messages
+          visibility_timeout message_group_id
+        ].freeze
 
         # Don't use this method directly: Configuration is a singleton class, use
         # +Aws::ActiveJob::SQS.config+ to access the singleton config.
         #
+        # This class provides a Configuration object for AWS ActiveJob
+        # by pulling configuration options from Runtime, the ENV, a YAML file,
+        # and default settings, in that order. Values set on queues are used
+        # preferentially to global values.
+        #
+        # # Environment Variables
+        # The Configuration loads global and queue specific values from your
+        # environment. Global keys take the form of:
+        # `AWS_ACTIVE_JOB_SQS_<KEY_NAME>` and queue specific keys take the
+        # form of: `AWS_ACTIVE_JOB_SQS_<QUEUE_NAME>_<KEY_NAME>`.
+        # <QUEUE_NAME> is case-insensitive and is always down cased. Configuring
+        # non-snake case queues (containing upper case) through ENV is
+        # not supported.
+        # Example:
+        #
+        #     export AWS_ACTIVE_JOB_SQS_MAX_MESSAGES = 5
+        #     export AWS_ACTIVE_JOB_SQS_DEFAULT_URL = https://my-queue.aws
+        #
         # @param [Hash] options
-        # @option options [Hash[Symbol, String]] :queues A mapping between the
-        #   active job queue name and the SQS Queue URL. Note: multiple active
-        #   job queues can map to the same SQS Queue URL.
+        # @option options [Hash[Symbol, Hash]] :queues A mapping between the
+        #   active job queue name and the queue properties. Valid properties
+        #   are: url [Required], max_messages, shutdown_timeout,
```

Comment:
This list can get dated. Maybe use a yard reference or point somewhere? Maybe a supported nested keys constant? File and env specific information belong on the class and not on initialize so this shouldn't be here anyway

Reply:
Moved some info to class, but keeping appropriate initialize options docs as well.